### ISSUE TITLE
[WC 4.4 beta1] Stats overview: no data shown after the order had been placed and completed

### ISSUE THREAD

**Issue by juliaamosova** [woocommerce/woocommerce-admin#4887]
**Describe the bug**

`Stats overview` widget shows no data after the order had been placed and completed. The data is, however, shown in the Analytics section if I click on `View detailed stats` at the bottom of the `Stats overview` widget. 

**To Reproduce**

1. Complete OBW.
2. Complete all tasks on the task list on the Home screen. 
3. Place an order on the site. 

I have an order in `Processing` status at this point but the `Stats overview` widget shows no orders:

![sale](https://user-images.githubusercontent.com/19143190/88851356-a9bbc980-d1ba-11ea-85b1-65cee16b344b.jpg)

4. Proceed with fulfilling the order and marking it as `Completed`. 
5. Check the `Stats overview` widget - still shows that no orders or sales had been made today:

![sale2](https://user-images.githubusercontent.com/19143190/88851542-e4bdfd00-d1ba-11ea-9e66-bcd888ef075f.jpg)

6. Click on `View detailed stats` at the bottom of the `Stats overview` widget - you should be taken to the `Analytics` section and see that the order was recorded there in the `Charts` section but not in the `Performance`: 

![sale3](https://user-images.githubusercontent.com/19143190/88851872-6ca40700-d1bb-11ea-83bb-706deb81efc6.jpg)

**Expected behavior**

Not sure if it takes some time for the data to be available but it is odd to me that `Charts` already showing some data but the `Stats overview` widget and the `Performance` section don't. 

**Screenshots**

See above. 

**Desktop (please complete the following information):**

* OS: macOS Catalina 10.15.5
* Browser: Chrome
* Version: 84.0.4147.89 (Official Build) (64-bit)

**Additional context**

N/A.

**Comment by timmyc** [comment]
@juliaamosova if you still have access to this site, can you see if there are any pending ActionScheduler jobs? I know @pmcpinto hit a similar scneario on a Pressable test site, and he had a fatal error showing on the ActionScheduler tab in the WooCommerce Settings. I haven't had a chance to dig into his site yet to see the cause there, but what you are describing sounds quite similar.

**Comment by juliaamosova** [comment]
@timmyc yes, I still have access to that site - it was a brand new JN site if anything. And I can also reproduce it consistently on JN sites. 

Today, the `Stats overview` widget still shows zero sales and zero orders:

![stats1](https://user-images.githubusercontent.com/19143190/88974387-8b230480-d286-11ea-9ba0-90db237a2ab4.jpg)

The Performance section of the `Analytics` overview, however, shows the data (it wasn't shown yesterday at the time of filing the bug):

![stats2](https://user-images.githubusercontent.com/19143190/88974396-8fe7b880-d286-11ea-90fc-832ad9a5c159.jpg)

No pending actions right now:

![stats3](https://user-images.githubusercontent.com/19143190/88974412-94ac6c80-d286-11ea-841f-a96e47713daa.jpg)

**Comment by juliaamosova** [comment]
Actually, today I had no orders on that site so the info on the `Stats Overview / Today` is correct. 

There is an order shown if I switch to `Stats Overview / Week to date`:

<img width="1438" alt="Screen Shot 2020-07-30 at 5 04 15 PM" src="https://user-images.githubusercontent.com/19143190/88974603-e35a0680-d286-11ea-9344-2bfbba723151.png">

So it looks good. 

Let me test again on another JN site and see if there will be any pending Action Scheduler actions once I place an order if the widget won't be updating with today's sales.

**Comment by becdetat** [comment]
I'm pretty sure this is a timezone issue. I can reproduce this and I'm at UTC+10 so the problem is pretty evident. I just placed an order in my JN site and it shows up for yesterday rather than today.

Here's the revenue chart for "today vs yesterday":

![image.png](https://images.zenhubusercontent.com/5e4207fe3e11c3c09f1c2919/d084eb1f-85bb-43c1-a7ee-e7c4298c3c58)

The blue line is today, the green line is yesterday.

Similarly if I set the stats overview to be week to date it includes the order:

![image.png](https://images.zenhubusercontent.com/5e4207fe3e11c3c09f1c2919/73fc3dae-58d5-4b81-91d6-f7dc98c38e40)

I'm not sure what to do about this as it seems to be working as designed - the order which I just placed was in fact "yesterday" in UTC-0. I would say it's definitely not expected behaviour to anybody in a negative UTC offset though.

**Comment by roykho** [comment]
So I am able to reproduce this issue. For me, it didn't matter which Date Range filtering I used, the `Performance` section just shows nothing was ordered. ![](https://d.pr/i/bR3MxV+)

It shows correctly in `Analytics->orders` though. ![](https://d.pr/i/gqdB8G+)

**Comment by joelclimbsthings** [comment]
Just wanted to add that I'm still seeing this issue on WC-Admin 2.1 beta. Potentially a timezone issue, although I'm at UTC - 8 so I don't see anything at all. The mildly befuddling thing to me is that the data on all other reports is correct; just not the Home Screen Stats and Analytics -> Overview -> Performance.

**Homescreen**
![image](https://user-images.githubusercontent.com/444632/108924536-ab822a00-75ef-11eb-93b5-1cfc8003e898.png)

**Analytics -> Overview**
![image](https://user-images.githubusercontent.com/444632/108924595-c5237180-75ef-11eb-88d0-6c8f0f8b8fab.png)

**Analytics -> Products**
![image](https://user-images.githubusercontent.com/444632/108924658-e6845d80-75ef-11eb-9926-bce232a2e996.png)


The Orders & Revenue reports all report correctly.

**Comment by psealock** [comment]
Estimate - 3. This smells like a timezone issue to me. I'm guessing this is 2.99 finding the bug 0.01 to make the actual fix.

**Comment by louwie17** [comment]
A user also had an issue when deleting orders that the revenue stats remained, I wonder if this could be related to this issue - https://github.com/woocommerce/woocommerce-admin/issues/6615

**Comment by chihsuan** [comment]
@mattsherman This bug is over 6 months old. But if Mothra can handle this, can I move this into "Mothra backlog"?

**Comment by chihsuan** [comment]
@mattsherman I just moved it to the Mothra backlog.
